unit_tests/src/test_rc_switch_user.cpp

```diff
@@ -282,9 +309,216 @@ TEST_CASE("rc_switch_user cannot be invoked by non-admins")
+{
+    load_client_api_plugins();
+
+    rodsEnv env;
+    _getRodsEnv(env);
+
+    //
+    // IMPORTANT: This test must be run using a rodsadmin account.
+    //
+
+    // clang-format off
+    namespace adm = irods::experimental::administration;
+    namespace fs  = irods::experimental::filesystem;
+    // clang-format on
+
+    irods::experimental::client_connection conn;
+
+    // Create a ticket that allows users to read the rodsadmin's home collection.
+    auto ticket = adm::ticket::client::create_ticket(conn, adm::ticket::ticket_type::read, env.rodsHome);
+    CHECK_FALSE(ticket.empty());
+
+    // Create a rodsuser.
+    const adm::user test_user_1{"rc_switch_user_test_user_1"};
+
+    REQUIRE_NOTHROW(adm::client::add_user(conn, test_user_1));
+
+    irods::at_scope_exit remove_user_1{
+        [&conn, &test_user_1] { CHECK_NOTHROW(adm::client::remove_user(conn, test_user_1)); }};
+
+    adm::user_password_property password_prop{"rods"};
+    REQUIRE_NOTHROW(adm::client::modify_user(conn, test_user_1, password_prop));
+
+#if 1
+    // Create a second rodsuser.
+    const adm::user test_user_2{"rc_switch_user_test_user_2"};
+
+    REQUIRE_NOTHROW(adm::client::add_user(conn, test_user_2));
+
+    irods::at_scope_exit remove_user_2{
+        [&conn, &test_user_2] { CHECK_NOTHROW(adm::client::remove_user(conn, test_user_2)); }};
+
+    REQUIRE_NOTHROW(adm::client::modify_user(conn, test_user_2, password_prop));
+#endif
+
+    // Connect to the iRODS server as test_user_1, with the rodsadmin acting as a proxy.
+    irods::experimental::fully_qualified_username proxy_user{env.rodsUserName, env.rodsZone};
+    irods::experimental::fully_qualified_username user{env.rodsUserName, env.rodsZone};
+    irods::experimental::client_connection test_user_conn{
+        irods::experimental::defer_authentication, env.rodsHost, env.rodsPort, proxy_user, user};
+    REQUIRE(clientLoginWithPassword(static_cast<RcComm*>(test_user_conn), password_prop.value.data()) == 0);
+
+    // Enable the ticket and show that test_user_1 can read the rodsadmin's home collection.
+    {
+        TicketAdminInput input{};
+        input.arg1 = "session";
+        input.arg2 = ticket.data();
+        input.arg3 = "";
+
+        REQUIRE(rcTicketAdmin(static_cast<RcComm*>(test_user_conn), &input) == 0);
+        CHECK(fs::client::exists(test_user_conn, env.rodsHome));
+    }
+
+    // FIXME Does modifying the password of the second user fail because of the additional client connection
+    // updating the session signature?
+    //
+    // I bet it does.
+#if 0
+    // Create a second rodsuser.
+    const adm::user test_user_2{"rc_switch_user_test_user_2"};
+
+    REQUIRE_NOTHROW(adm::client::add_user(conn, test_user_2));
+
+    irods::at_scope_exit remove_user_2{
+        [&conn, &test_user_2] { CHECK_NOTHROW(adm::client::remove_user(conn, test_user_2)); }};
+
+    REQUIRE_NOTHROW(adm::client::modify_user(conn, test_user_2, password_prop));
+#endif
```

Comment:
Enable once #7192 is resolved.

Reply:
Restored and confirmed to work.

Solution: Modified the user administration library by replacing the use of `obfEncodeByKeyV2` with `obfEncodeByKey`. This change aligns with how `iadmin` works.

Reply:
Why does V2 exist?   Presumably it was added for a good reason?

Reply:
I haven't investigated why it exists, but it is only used in iadmin, in one place. Nothing in the server uses it.

It seems to exist as a stronger version of the original, with no obvious insight into why it was introduced.

Reply:
https://github.com/irods/irods-legacy/commit/0cb144cbd4c9bc57b31f834ab38bc8eb73b72ab3

Reply:
Nice. V2 presents a problem though. It's not clear to me if there's a session signature available on the server-side. I certainly couldn't find anything in the codebase.

Without that, native auth password management doesn't seem doable from within the server.